File: 1414-find-the-minimum-number-of-fibonacci-numbers-whose-sum-is-k/1414-find-the-minimum-number-of-fibonacci-numbers-whose-sum-is-k.js
/**
 * @param {number} k
 * @return {number}
 */
var findMinFibonacciNumbers = function (k) {

    let fib = [1, 1];

    for (let i = 2; fib[fib.length-1] <= k; i++) {
        fib[i] = fib[i - 1] + fib[i - 2];
    }

    let count = 0;

    let kk = k;

    let i = fib.length-1;
    while (kk != 0) {
        if (fib[i] <= kk) {
            count++;
            kk = kk - fib[i];
        }
        i--;
    }
 
    return count;
};

File: 1414-find-the-minimum-number-of-fibonacci-numbers-whose-sum-is-k/1414-find-the-minimum-number-of-fibonacci-numbers-whose-sum-is-k.java
class Solution {
    public int findMinFibonacciNumbers(int k) {
         List<Integer> fib = new ArrayList<>(List.of(1,1));

    for (int i = 2; fib.get(fib.size() - 1) <= k; i++) {
        fib.add(fib.get(i-1)+fib.get(i-2));
    }

    int count = 0;

    int kk = k;

    int i = fib.size()- 1;
    while (kk != 0) {
        if (fib.get(i) <= kk) {
            count++;
            kk = kk - fib.get(i);
        }
        i--;
    }
 
    return count;
    }
}